src/libsyntax_ext/test.rs

```diff
@@ -53,7 +53,7 @@ pub fn expand_test_or_bench(
         if let Annotatable::Item(i) = item { i }
         else {
             cx.parse_sess.span_diagnostic.span_fatal(item.span(),
-                "#[test] attribute is only allowed on fn items").raise();
+                "#[test] attribute is only allowed on non associated functions").raise();
```

Comment:
I would rephrase `non associated functions` as `free functions` to avoid the negative.

Reply:
a simple search of "rust free functions" on duckduckgo or "free functions" on the rust book has no results while "associated functions" gives you the expected page, that's why I chose that phrase.

Reply:
Sadly the [page in the book](https://doc.rust-lang.org/book/2018-edition/ch05-03-method-syntax.html) is [quite wrong](https://doc.rust-lang.org/stable/reference/items/associated-items.html) in that it defines *"associated function"* as a non-method inside an implementation (trait/inherent). Thus, by saying "non-associated function", by the incorrect definition in the book you are stating that `#[test]` is allowed on methods. Given this, I think if you want to impart a correct error you need to say something like:

> `"#[test] attribute is only allowed on free functions without arguments"`

or:

> `"#[test] attribute is only allowed on no-argument functions outside implementations"`